devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy

```diff
@@ -0,0 +1,250 @@
+		}
+		standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label:'reset kWh', action:"reset"
+		}
+
+		main(["switch"])
+		details(["switch","power","energy","refresh","reset"])
+	}
+}
+
+def parse(String description) {
+	def result = null
+	if (description.startsWith("Err")) {
+		result = createEvent(descriptionText:description, isStateChange:true)
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd, null)
+		}
+	}
+	log.debug "parsed '${description}' to ${result.inspect()}"
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, ep) {
+	log.debug "Multichannel command ${cmd}" + (ep ? " from endpoint $ep" : "")
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep) {
+	log.debug "Basic ${cmd}" + (ep ? " from endpoint $ep" : "")
+	def value = cmd.value ? "on" : "off"
+	ep ? changeSwitch(ep, value) : []
+}
+
+private changeSwitch(endpoint, value) {
+	def result = []
+	if(endpoint == 1) {
+		result += createEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+	} else {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		child?.sendEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep) {
+	log.debug "Meter ${cmd}" + (ep ? " from endpoint $ep" : "")
+	def result = []
+	if(ep == 1) {
+		result += createEvent(createMeterEventMap(cmd))
+	} else if(ep) {
+		String childDni = "${device.deviceNetworkId}:$ep"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		child?.sendEvent(createMeterEventMap(cmd))
+	} else {
+		result += zwaveEvent(cmd)
+	}
+	result += response(encap(zwave.meterV3.meterGet(scale: 2), ep))
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def event = createEvent([isStateChange:  false, descriptionText: "Wattage change has been detected. Refreshing each endpoint"])
+	[event, response(refreshAll())]
+}
+
+private createMeterEventMap(cmd) {
+	def eventMap = [:]
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			eventMap.name = "energy"
+			eventMap.value = cmd.scaledMeterValue
+			eventMap.unit = "kWh"
+		} else if (cmd.scale == 2) {
+			eventMap.name = "power"
+			eventMap.value = Math.round(cmd.scaledMeterValue)
+			eventMap.unit = "W"
+		}
+	}
+	eventMap
```

Comment:
if you call `createEvent()` here you could save yourself from wrapping it above.

Reply:
Wouldn't it affect sending event to child device (line 102)?